src/test/common/commandExecutor.test.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {CommandExecutor} from "../../common/commandExecutor";
+import {Log} from "../../common/log/log";
+import {ChildProcess} from "child_process";
+
+import * as assert from "assert";
+import * as semver from "semver";
+import * as sinon from "sinon";
+import * as Q from "q";
+
+suite("commandExecutor", function() {
+    suite("commonContext", function() {
+        teardown(function() {
+            let mockedMethods = [Log.logMessage, Log.logCommandStatus];
+
+            mockedMethods.forEach((method) => {
+                if (method.hasOwnProperty("restore")) {
+                    (<any>method).restore();
+                }
+            });
+        });
+
+        test("should execute a command", function() {
+            let ce = new CommandExecutor();
```

Comment:
all the test have this line. Maybe we should put it in a setUp, or use a constant suite variable if you are happy assuming that this class has no state.